retrystorage/retry_storage.go

```diff
@@ -0,0 +1,194 @@
+package retrystorage
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/cenkalti/backoff"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/giantswarm/microstorage"
+)
+
+const (
+	// defaultMaxAttempts is the default number of attempts of each storate
+	// operations before it finally fails.
+	defaultMaxAttempts = 3
+)
+
+type stoppingBackOff struct {
```

Comment:
I wanted to provide "sensible default" in `DefaultConfig`. I.e. I don't want to create a backoff configured in the same way in all our services. That's why I find `DefaultConfig` powerful. I don't think backoff that retires forever is a sensible default. Do you?